x/name/types/msgs.go

```diff
@@ -84,6 +96,32 @@ func NewMsgModifyNameRequest(authority string, name string, owner sdk.AccAddress
 }
 
 // ValidateBasic runs stateless validation checks on the message.
+func (msg MsgCreateRootNameRequest) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Authority); err != nil {
+		return ErrInvalidAddress
+	}
+
+	if _, err := sdk.AccAddressFromBech32(msg.Record.Address); err != nil {
+		return ErrInvalidAddress
+	}
+
+	if strings.TrimSpace(msg.Record.Name) == "" {
```

Comment:
Name record has a `ValidateBasic` method that does these checks. You could probably call that instead. It doesn't check the Address though.  Maybe it should?

While you are at it.  You could probably name the `ValidateBasic` method of NameRecord to `Validate`.  Since we just want to reserve `ValidateBasic` to types that implement the sdk.Msg interface.